### ISSUE TITLE
Consider making web5-kt a multiplatform library

### ISSUE THREAD

**Issue by andresuribe87** [TBD54566975/web5-kt#145]
Recently, I've been learning more about https://www.jetbrains.com/kotlin-multiplatform

tl;dr; we can share code across iOS, Android, JVM, and JS by writing it in Kotlin.

The obvious advantage over other alternatives is that we already have written a vast a amount of code in Kotlin that's been tested.

I think it would be useful to do a spike on how this works before we make a decision on whether rust is the preferred way to move forwards. In the spike, we should look to answer the following questions:
* What does it look like to consume the multiplatform library from Android, iOS, and JS?
* What are the limitations of using such a mechanism?
* Are there any dependencies that wouldn't be available? (for example, it's possible that jackson doesn't play well with multi-platform, among others).
